zebra-utils/src/bin/scanning-results-reader/main.rs

```diff
@@ -0,0 +1,128 @@
+//! Displays Zebra's scanning results:
+//!
+//! 1. Opens Zebra's scanning storage and reads the results containing scanning keys and TXIDs.
+//! 2. Fetches the transactions by their TXIDs from Zebra using the `getrawtransaction` RPC.
+//! 3. Decrypts the tx outputs using the corresponding scanning key.
+//! 4. Prints the memos in the outputs.
+
+use std::collections::HashMap;
+
+use jsonrpc::simple_http::SimpleHttpTransport;
+use jsonrpc::Client;
+
+use serde_json::value::RawValue;
+use zcash_client_backend::decrypt_transaction;
+use zcash_client_backend::keys::UnifiedFullViewingKey;
+use zcash_primitives::consensus::{BlockHeight, BranchId};
+use zcash_primitives::transaction::Transaction;
+use zcash_primitives::zip32::AccountId;
+use zebra_scan::scan::sapling_key_to_scan_block_keys;
+use zebra_scan::{storage::Storage, Config};
+
+/// Prints the memos of transactions from Zebra's scanning results storage.
+///
+/// Reads the results storage, iterates through all decrypted memos, and prints the them to standard
+/// output. Filters out some frequent and uninteresting memos typically associated with ZECPages.
+///
+/// Notes:
+///
+/// - `#[allow(clippy::print_stdout)]` is set to allow usage of `println!` for displaying the memos.
+/// - This function expects Zebra's RPC server to be available.
+///
+/// # Panics
+///
+/// When:
+///
+/// - The Sapling key from the storage is not valid.
+/// - There is no diversifiable full viewing key (dfvk) available.
+/// - The RPC response cannot be decoded from a hex string to bytes.
+/// - The transaction fetched via RPC cannot be deserialized from raw bytes.
+#[allow(clippy::print_stdout)]
+pub fn main() {
+    let network = zcash_primitives::consensus::Network::MainNetwork;
+    let storage = Storage::new(&Config::default(), zebra_network(&network), true);
+    let mut prev_memo = "".to_owned();
```

Comment:
Optional Style: Strictly we should use `None` for a missing value, but that might complicate the code a lot.

This doesn't change functionality and is probably fine for a basic tool, but it might help to add a comment.

```suggestion
    // If the first memo is empty, it doesn't get printed. But we never print empty memos anyway.
    let mut prev_memo = "".to_owned();
```

Reply:
Yes, this will keep excluding empty memos until the first non-empty one. I kept many things basic when writing this.